Sample/ECommerce.Equinox/ECommerce.Api/Program.fs

```diff
@@ -0,0 +1,179 @@
+module ECommerce.Api.Program
+
+open ECommerce
+open Microsoft.AspNetCore.Hosting
+open Microsoft.Extensions.DependencyInjection
+open Serilog
+open System
+
+exception MissingArg of message : string with override this.Message = this.message
+
+type Configuration(tryGet) =
+
+    let get key =
+        match tryGet key with
+        | Some value -> value
+        | None -> raise (MissingArg $"Missing Argument/Environment Variable %s{key}")
+
+    member _.CosmosConnection =             get "EQUINOX_COSMOS_CONNECTION"
+    member _.CosmosDatabase =               get "EQUINOX_COSMOS_DATABASE"
+    member _.CosmosContainer =              get "EQUINOX_COSMOS_CONTAINER"
+
+    member _.EventStoreHost =               get "EQUINOX_ES_HOST"
+    member _.EventStorePort =               tryGet "EQUINOX_ES_PORT" |> Option.map int
+    member _.EventStoreUsername =           get "EQUINOX_ES_USERNAME"
+    member _.EventStorePassword =           get "EQUINOX_ES_PASSWORD"
+
+module Args =
```

Comment:
**COMMENT:** F# is great, but this configuration part looks wild! 😅

Reply:
Yes, Argu is definitely its own language / box of frogs. In general I like how it lets you express all the dependencies of an app concisely, and the `--help` it produces is pretty comprehensive. But it definitely has its haters!
- http://fsprojects.github.io/Argu/tutorial.html
- https://github.com/fsprojects/Argu/tree/master/samples/Argu.Samples.LS